igmspec/ingest/cos_halos.py

```diff
@@ -0,0 +1,207 @@
+    meta.add_column(Column(meta_IDs, name='IGM_ID'))
+
+    # Build spectra (and parse for meta)
+    nspec = len(meta)
+    max_npix = 20000  # Just needs to be large enough
+    data = np.ma.empty((1,),
+                       dtype=[(str('wave'), 'float64', (max_npix)),
+                              (str('flux'), 'float32', (max_npix)),
+                              (str('sig'),  'float32', (max_npix)),
+                              #(str('co'),   'float32', (max_npix)),
+                             ])
+    # Init
+    spec_set = hdf[sname].create_dataset('spec', data=data, chunks=True,
+                                         maxshape=(None,), compression='gzip')
+    spec_set.resize((nspec,))
+    wvminlist = []
+    wvmaxlist = []
+    npixlist = []
+    speclist = []
+    # Loop
+    path = os.getenv('RAW_IGMSPEC')+'/COS-Halos/'
+    maxpix = 0
+    for jj,row in enumerate(meta):
+        # Generate full file
+        coord = ltu.radec_to_coord((row['RA'],row['DEC']))
+        full_file = path+'/J{:s}{:s}_nbin3_coadd.fits.gz'.format(coord.ra.to_string(unit=u.hour,sep='',pad=True)[0:4],
+                                               coord.dec.to_string(sep='',pad=True,alwayssign=True)[0:5])
+        # Extract
+        print("COS-Halos: Reading {:s}".format(full_file))
+        spec = lsio.readspec(full_file)
+        # Parse name
+        fname = full_file.split('/')[-1]
+        # npix
+        npix = spec.npix
+        if npix > max_npix:
+            raise ValueError("Not enough pixels in the data... ({:d})".format(npix))
+        else:
+            maxpix = max(npix,maxpix)
+        # Some fiddling about
+        for key in ['wave','flux','sig']:
+            data[key] = 0.  # Important to init (for compression too)
+        data['flux'][0][:npix] = spec.flux.value
+        data['sig'][0][:npix] = spec.sig.value
+        data['wave'][0][:npix] = spec.wavelength.value
+        # Meta
+        speclist.append(str(fname))
+        wvminlist.append(np.min(data['wave'][0][:npix]))
+        wvmaxlist.append(np.max(data['wave'][0][:npix]))
+        npixlist.append(npix)
+        if chk_meta_only:
+            continue
+        # Only way to set the dataset correctly
+        spec_set[jj] = data
+
+    #
+    print("Max pix = {:d}".format(maxpix))
+    # Add columns
+    meta.add_column(Column(speclist, name='SPEC_FILE'))
+    meta.add_column(Column(npixlist, name='NPIX'))
+    meta.add_column(Column(wvminlist, name='WV_MIN'))
+    meta.add_column(Column(wvmaxlist, name='WV_MAX'))
+    meta.add_column(Column(np.arange(nspec,dtype=int), name='SURVEY_ID'))
+
+    # Add HDLLS meta to hdf5
+    if iiu.chk_meta(meta):
+        if chk_meta_only:
+            pdb.set_trace()
+        hdf[sname]['meta'] = meta
+    else:
+        raise ValueError("meta file failed")
+    # References
+    refs = [dict(url='http://adsabs.harvard.edu/abs/2013ApJ...777...59T',
+                 bib='tumlinson+13')
+            ]
```

Comment:
You might want to add Werk+12 reference and Werk+13 reference as well, since that's the metal lines and galaxy databases.